client/blocks/login/index.jsx

```diff
@@ -69,6 +69,7 @@ class Login extends Component {
 		twoFactorEnabled: PropTypes.bool,
 		twoFactorNotificationSent: PropTypes.string,
 		isSecurityKeySupported: PropTypes.bool,
+		userEmail: PropTypes.string,
```

Comment:
Where is `userEmail` coming from?

Reply:
Normally it's coming from the page's querystring ([see here](https://github.com/Automattic/wp-calypso/pull/41799/files#diff-29356c8042a170020e49bae30afff806R677-R678)). So you'll land in the login flow with an URL like `/login?...&email_address=foo@bar.com`, and the `Email` field will be prefilled with `foo@bar.com`.

What I've made in this PR is to have the possibility to override that value. So the `LoginBlock` could be rendered like:
```js
<LoginBlock userEmail="asdf@aaa.com" />
```
And the prefilled Email field will be `asdf@aaa.com`, regardless of the URL the user landed on. Because the whole login/signup/authorize Woo flow is implemented in a single page (without the URL ever changing), this was a needed addition.